biorxiv/word_vector_experiment/biorxiv_generate_embeddings.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Generate BioRxiv Document Embeddings 
+
+# This notebook is designed to generate document embeddings for every article in bioRxiv.
+
+# In[2]:
+
+
+from pathlib import Path
+import os
+
+from gensim.models import Word2Vec
+from gensim.parsing.preprocessing import remove_stopwords
+import lxml.etree as ET
+import pandas as pd
+from sklearn.manifold import TSNE
+from tqdm import tqdm_notebook
+import umap
+
+
+# # Output Documents to File
+
+# This section dumps all of biorxiv text into a single document in order to train the word2vec model. This is for ease of training the model.
+
+# In[3]:
+
+
+journal_map_df = pd.read_csv("../exploratory_data_analysis/output/biorxiv_article_metadata.tsv", sep="\t")
+journal_map_df.head()
+
+
+# In[ ]:
+
+
+parser = ET.XMLParser(encoding='UTF-8', recover=True)
+
+# Only use the most current version of the documents
+latest_journal_version = (
+    journal_map_df.groupby("doi")
+    .agg({"document":"first", "doi":"last"})
+)
+
+with open("output/word2vec_input/biorxiv_text.txt", "w") as f:
+    for idx, article in tqdm_notebook(latest_journal_version.iterrows()):
+        tree = (
+            ET.parse(
+                open(f"../biorxiv_articles/{os.path.basename(article['document'])}", "rb"),
+                parser = parser
+            )
+        )
+        
+        root = tree.getroot()
+        
+        # Grab the abstract text
+        abstract_text = root.xpath("//abstract/*//text()")
+        abstract_text = list(map(lambda x: remove_stopwords(x), abstract_text))
+        f.write("".join(abstract_text))
+        
+        # Grab the body text
+        article_text = root.xpath("//body/sec/*//text()")
+        article_text = list(map(lambda x: remove_stopwords(x), article_text))
+        f.write("".join(article_text))
+        
+        #Sign of a new article
+        f.write("\n\n")
+
+
+# # Train Word2Vec
+
+# This section trains the word2vec model (continuous bag of words [CBOW]). Since the number of dimensions can vary I decided to train multiple models: 150, 250, 300. Each model is saved into is own respective directory.
+
+# In[ ]:
+
+
+class DocIterator: 
+    def __init__(self, filepath): 
+        self.filepath = filepath 
+
+    def __iter__(self): 
+        for line in open(self.filepath, "r"): 
+            yield line.split() 
+
+
+# In[ ]:
+
+
+word_embedding_sizes = [150, 250, 300]
```

Comment:
This is the number of features used to represent a given word.  So "apple" is represented by a vector of size 150x1 ?

Reply:
Bingo.  "apple" is mapped to a 1x150 vector of network weights.